clients/java/src/test/java/io/camunda/zeebe/client/usertask/CompleteUserTaskTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.usertask;
+
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.protocol.rest.ProblemDetail;
+import io.camunda.zeebe.client.protocol.rest.UserTaskCompletionRequest;
+import io.camunda.zeebe.client.util.ClientRestTest;
+import io.camunda.zeebe.client.util.RestGatewayService;
+import org.junit.jupiter.api.Test;
+
+public final class CompleteUserTaskTest extends ClientRestTest {
+
+  @Test
+  void shouldCompleteUserTask() {
+    // when
+    client.newUserTaskCompleteCommand(123L).send().join();
+
+    // then
+    final UserTaskCompletionRequest request =
+        gatewayService.getLastRequest(UserTaskCompletionRequest.class);
+    assertThat(request.getAction()).isNull();
+    assertThat(request.getVariables()).isNull();
+  }
+
+  @Test
+  void shouldCompleteUserTaskWithAction() {
+    // when
+    client.newUserTaskCompleteCommand(123L).action("foo").send().join();
+
+    // then
+    final UserTaskCompletionRequest request =
+        gatewayService.getLastRequest(UserTaskCompletionRequest.class);
+    assertThat(request.getAction()).isEqualTo("foo");
```

Comment:
💭 Unrelated: I hadn't seen the `action` property for these endpoints yet. What's the intended use case for them?

Reply:
Using this, users can define a custom user task lifecycle on top of the out-of-the-box lifecycle we provide. For example, they can have states like `override` and `complete` that both lead to a user task completion but have different semantics. They can inspect the custom action later in task listeners (when we have that) and analyze them in Optimize, for example.